scripts/globals/combat/physical_utilities.lua

```diff
@@ -428,6 +428,34 @@ xi.combat.physical.calculateMeleePDIF = function(actor, target, weaponType, wsAt
     return pDif
 end
 
+-- calculates melee critical rate, if true is passed, then only returns crit from dex vs agi
+xi.combat.physical.calculateMeleeCriticalRate = function(actor, target, nativeOnly)
+    -- See reference at https://www.bg-wiki.com/ffxi/Critical_Hit_Rate
+    local nativecrit = 0
```

Comment:
Shouldn't this be 0.05? That's the base crit rate -- the rest should be adding to it, no?

Reply:
The elseif chain below could also use some simplification.

```lua
xi.combat.physical.calculateMeleeCriticalRate = function(actor, target)
    -- See reference at https://www.bg-wiki.com/ffxi/Critical_Hit_Rate
    local criticalRate = 0.05
    local dDex         = actor:getStat(xi.mod.DEX) - target:getStat(xi.mod.AGI)

    if dDex > 50 then
        criticalRate = 0.15
    elseif dDex >= 40 then
        criticalRate = (dDex - 35) / 100
    elseif dDex >= 30 then
        criticalRate = 0.04
    elseif dDex >= 20 then
        criticalRate = 0.03
    elseif dDex >= 14 then
        criticalRate = 0.02
    elseif dDex >= 7 then
        criticalRate = 0.01
    end

    -- We should seriously consider adding merits, job points and other modifiers here. 
    
    return criticalRate
end
```

I also dont like that `nativeOnly` argument in there.
In which cases would we want the critical rate from dStat only?

Reply:
Yea the goal of this function was to be a drop-in replacement for what was used in `weaponskills.lua` as well as for use in `bluemagic.lua`. 

You might be right that we should include the base .05, i just copied the function over. Ditto for the chained if/else.

Reply:
On second thought, go with Xaver's function here

Reply:
Understood. I've tweaked as I think youl would want while keeping the functions separate for base and extra mods